### PR TITLE
Reenable menu shadow for reporting menu

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

I've also included some css tweaks . Instead of a padding the pagewarp now only uses a margin. That way that space won't be included in UI tests of the pagewrap anymore. This will require an update of a lot screenshots (as the free space will be removed), but we will no longer have problems with the menu shadow on screenshots.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by sgiehl** [comment]
The CSS tweaks I've added should help a lot around UI tests regarding the menu shadow being visible and also around the free space visible on some screens, when the menu is higher than the reports.
Note: Merging this would mean we need to update the sceenshots once in all plugins, but we would have less failures in plugins due to changes in the menu in the future.

@matomo-org/core-reviewers @AltamashShaikh @snake14 Would you mind taking a quick look at the changes and share your opinion?

File: plugins/CoreHome/stylesheets/layout.less
Comment (michalkleiner) [inline plugins/CoreHome/stylesheets/layout.less:208]: If we add `flex: none` here, we don't need to do any width setting on the `.pageWrap` at all (the calc, the 100% on mobile..)

```suggestion
    height: fit-content;
    flex: none;
```

**Comment by michalkleiner** [comment]
The changes in general look ok. As commented above, we can do `flex: none` on the lefthand menu and then we don't need to do any width calc on the pageWrap itself, it would always behave as needed.

I'm not a big fan of having values like 9px or 18px (not on 4, or even better, 8px grid), but that's a discussion for another time.

File: plugins/CoreHome/stylesheets/layout.less
Comment (sgiehl) [reply plugins/CoreHome/stylesheets/layout.less:208]: This doesn't have the expected effect. I will add it, as it ensure the menu won't be resized, but the width on the pagewrap os still needed. The problem is, that by default the pagewrap would only use the minimal required space. But it actually should take all space that is available. Otherwise the reports might not be as wide as they were before.
You can see that when removing the width properties and viewing e.g. a pages report. The report would then not stretch across the whole screen anymore

Comment (michalkleiner) [reply plugins/CoreHome/stylesheets/layout.less:208]: I'll recheck what I'd disabled when I was testing the PR in a browser.. might have been removing the calc but keeping 100% everywhere (definitely removing the `calc`).

Comment (sgiehl) [reply plugins/CoreHome/stylesheets/layout.less:208]: Only using 100% might now also work. The calc was needed, as 100% tried to consume too much space. But with flex: none, that might work now. I'll give it a try.

Comment (sgiehl) [reply plugins/CoreHome/stylesheets/layout.less:208]: I'll revert removing the calculated width again. There were too many unexpected changes in the width of reports and notifications. You can see the differences here:
with calculated width: https://builds-artifacts.matomo.org/matomo-org/matomo/menushadow/7245530051/
with 100% width: https://builds-artifacts.matomo.org/matomo-org/matomo/menushadow/7248776660/

Comment (michalkleiner) [reply plugins/CoreHome/stylesheets/layout.less:208]: @sgiehl I went over the majority of screenshots in the 'with 100% width' version andthe changes around notifications are actually improvements over what it is now.
Since we're changing so many screenshots already, I'd say the goal doesn't need to be necessarily to match all one for one, but small improvements in the CSS (like we have here) resulting in small improvements in the UI (like we may have here) should be fine.

Basically the main difference is that the notification behaves better with 100% width compared to the calc version (see screenshots below, don't mind the temporary red background used for easier visualisation).

With calc (as we have now - notice the slight indent inward)
notification with calc


With 100% width (better aligned, no indent), not functional change in the actual UI in browser.
notification with 100 width

I'd say it'd be better to have flex:none and width: 100% for the rest and update the additional UI tests as they better reflect the improvement, but I'd leave the final decision on this to you.

Comment (sgiehl) [reply plugins/CoreHome/stylesheets/layout.less:208]: Thought the same when looking at those. But there are also changes like:
https://builds-artifacts.matomo.org/matomo-org/matomo/menushadow/7248776660/UIIntegrationTest_actions_pages.png
https://builds-artifacts.matomo.org/matomo-org/matomo/menushadow/7248776660/GoalsPages_overview.png
https://builds-artifacts.matomo.org/matomo-org/matomo/menushadow/7248776660/PivotByDimension_pivoted.png
Those changes look unexpected and would need further investigation

Comment (michalkleiner) [reply plugins/CoreHome/stylesheets/layout.less:208]: Yeah, I agree, some of the tests may need to looked at manually in the UI. The line-height change doesn't make much sense to me, so might be a one off maybe, re-running the tests could help?
I can look into those in a PR on top of this PR. I guess we can iterate either here or in a separate PR until we decide to update UI tests in plugins and until we decide it's too much time for too little benefit.

Comment (sgiehl) [reply plugins/CoreHome/stylesheets/layout.less:208]: I gave removing the calculated width another try. The main problem is the reports by dimensions view for goals and referrers. As they add another floating, that breaks the flex layout, or at least the width. The element itself takes 100% space (ignoring the menu) and is getting too big then. I tried fixing it, by using flex there as well, but did find a quick fix for it. Imho we should go with this one as it is for now. Guess we would need to fully rework some parts of the layout to make that work correctly. And that's not really worth the effort only to get rid of a calculated width...

Comment (michalkleiner) [reply plugins/CoreHome/stylesheets/layout.less:208]: Thanks for looking into this again. I agree, we will hopefully get to addressing the UI from a bigger picture perspective at some point soon, so this is good for now.

**Review by michalkleiner — CHANGES_REQUESTED** [review]
Code change looks good. Some more UI test screenshots need updating.

**Comment by sgiehl** [comment]
I've now created PRs for the submodules where the UI tests were failing. Guess we should merge them and update the submodules in this PR before merging. The remaining failing UI tests are unrelated.

**Review by michalkleiner — APPROVED** [review]
This is good for me but would be good to get another 👀 from @matomo-org/core-reviewers

File: plugins/CoreHome/stylesheets/layout.less
Comment (mneudert) [inline plugins/CoreHome/stylesheets/layout.less]:
```suggestion
      width: calc(~'100% - 18px') !important; // 100% - pageWrap margin
```

_(same problem as `max-width: 992px`)_

File: plugins/CoreHome/stylesheets/layout.less
Comment (mneudert) [inline plugins/CoreHome/stylesheets/layout.less]:
```suggestion
      width: calc(~'100% - 18px') !important; // 100% - pageWrap margin
```

Or we need to replace the margin with more padding for those resolutions. Otherwise the content will be stretched to the scrollbar (and create a horizontal scrollbar):



It should look like this with the change:



_Yes, the navigation elements are already misaligned with their custom margins. That was not changed here._

File: tests/UI/specs/UIIntegration_spec.js
Comment (mneudert) [inline tests/UI/specs/UIIntegration_spec.js:61]: Both `CustomAlerts` and `CustomVariables` have this hack in one place.

_Mentioning it here to keep everything in one place._

**Review by mneudert — CHANGES_REQUESTED** [review]

File: plugins/CoreHome/stylesheets/layout.less
Comment (sgiehl) [reply plugins/CoreHome/stylesheets/layout.less]: The calculation makes sense. Interestingly it looks the same for me. Which browser did you check that with?

Comment (mneudert) [reply plugins/CoreHome/stylesheets/layout.less]: Both Firefox 120 and Chrome 120.

Just a caveat: the visual problem only occurs if the datatable (the screenshots are from "Behaviour > Pages") has content. If it is empty it does not stretch to the scrollbar (but also should not begin to do that with the change).

Comment (sgiehl) [reply plugins/CoreHome/stylesheets/layout.less]: Ah. Yes. There it is reproducible.

File: tests/UI/specs/UIIntegration_spec.js
Comment (sgiehl) [reply tests/UI/specs/UIIntegration_spec.js:61]: removed that in the submodule PRs.

**Review by mneudert — APPROVED** [review]

**Comment by sgiehl** [comment]
@AltamashShaikh @snake14 I'm going to merge this PR now. It will most likely cause a lot of failures in UI tests of plugins. All tests taking a screenshot of the pageWrap element will get (a bit) smaller. But the changes in this PR will also prevent that parts of the menu shadow will be visible on such screenshots (so any hacks around that will no longer be needed). Also the height of the pageWarp element isn't impacted by the menu height anymore. This should make those UI tests a bit more stable and more independent from changes in core.

**Comment by snake14** [comment]
> @AltamashShaikh @snake14 I'm going to merge this PR now. It will most likely cause a lot of failures in UI tests of plugins. All tests taking a screenshot of the pageWrap element will get (a bit) smaller. But the changes in this PR will also prevent that parts of the menu shadow will be visible on such screenshots (so any hacks around that will no longer be needed). Also the height of the pageWarp element isn't impacted by the menu height anymore. This should make those UI tests a bit more stable and more independent from changes in core.

Thank you @sgiehl . That's great news. I've been hoping this would be merged soon. I'll start updating the plugins.

**Comment by snake14** [comment]
Oh. @sgiehl Are these changes going to be back-ported to 4.x? We currently have several plugins with failing builds due to the navigation shadow.

**Comment by sgiehl** [comment]
@snake14 We will most likely only backport security fixes to 4.x-dev. So this won't be backported.

**Comment by snake14** [comment]
> @snake14 We will most likely only backport security fixes to 4.x-dev. So this won't be backported.

@sgiehl Just thought I'd check since the marketplace changes that broke the builds was backported. I'll simply update the 4.x screenshots if this fix isn't going to be applied to that branch.

**Pull request merged**
